src/main/scala/co/teapot/graph/TransposedGraphView.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2016 Teapot, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package co.teapot.graph
+
+/**
+  * A transposed view of a given graph.
+  */
+private[graph] class TransposedGraphView(graph: DirectedGraph) extends DirectedGraph {
+  override def outNeighbors(id: Int): IndexedSeq[Int] =
+    graph.inNeighbors(id)
+  override def inNeighbors(id: Int): IndexedSeq[Int] =
+    graph.outNeighbors(id)
+
+  override def outNeighbor(id: Int, i: Int): Int =
+    graph.inNeighbor(id, i)
+  override def inNeighbor(id: Int, i: Int): Int =
+    graph.outNeighbor(id, i)
+
+  override def outDegree(id: Int): Int = graph.inDegree(id)
+  override def inDegree(id: Int): Int = graph.outDegree(id)
+
+  // These just defer to the underlying graph.
+  override def maxNodeId: Int = graph.maxNodeId
```

Comment:
I agree with this completely.  I just implemented a GraphView class which forwards all methods, and I'll change the bidirectional view class in teapot compute to use it when I work on that code next.

I feel like there should be a way to generate GraphView without explicitly listing the methods, something like
```scala
class GraphView(graph: DirectedGraph) extends DirectedGraph {
  import graph._
}
```
But when I tried that compiler claims GraphView doesn't implement the methods of DirectedGraph.  I found a blog post referencing an old compiler plugin, but then I decided it isn't worth messing with a plugin.  The api of DirectedGraph should be pretty stable, and now the GraphView is in one place.